sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3825,10 +3852,70 @@ pi_result piextMemGetNativeHandle(pi_mem Mem, pi_native_handle *NativeHandle) {
 }
 
 pi_result piextMemCreateWithNativeHandle(pi_native_handle NativeHandle,
-                                         pi_mem *Mem) {
-  (void)NativeHandle;
-  (void)Mem;
-  die("piextMemCreateWithNativeHandle: not supported");
+                                         pi_context Context,
+                                         bool ownNativeHandle, pi_mem *Mem) {
+  PI_ASSERT(Mem, PI_INVALID_VALUE);
+  PI_ASSERT(NativeHandle, PI_INVALID_VALUE);
+  PI_ASSERT(Context, PI_INVALID_CONTEXT);
+  PI_ASSERT(Context->Devices.size() > 0, PI_INVALID_CONTEXT);
```

Comment:
Let's check for exactly 1 device meantime

Reply:
Idea was that if user provides pointer obtained from zeMemAllocHost then it is safe to have context with multiple devices.

Reply:
Fixed to check exactly 1 device.